### PR TITLE
Support filtering samples by batch number

### PR DESCRIPTION
**Pull request by vladsavelyev**
Useful for large-cohort datasets.

Batch IDs are set in the config, similar to `only_samples`:

```toml
[workflow]
only_batches = [1, 2]
```

### CODE REVIEW

File: cpg_utils/workflows/inputs.py [hunk 1]
```diff
@@ -161,6 +157,24 @@ def _populate_alignment_inputs(
     for sample in cohort.get_samples():
         for d in found_seqs_by_sid.get(sample.id, []):
             seq = Sequence.parse(d, check_existence=check_existence)
+            if only_batches:
+                if 'batch' not in seq.meta:
```
Comment (illusional) [inline cpg_utils/workflows/inputs.py:161]: Should probably come up with a list of these "patterns" around the place. Or even maybe define them in MM itself to make it clearer how these are used and provided.

Comment (vladsavelyev) [reply cpg_utils/workflows/inputs.py:161]: Agree, makes sense to do this in metamist!

Comment (lgruen) [reply cpg_utils/workflows/inputs.py:161]: It also took me a few seconds to realize that this is about _sample_ batches and completely unrelated to Hail batches. The term "batch number" is unfortunately somewhat overloaded. This becomes even more tricky as we also want to store Hail Batch related information in Metamist, e.g. resource usage.

Can we make this a little clearer by using `sample_batch` or similar instead?

**Review by lgruen — CHANGES_REQUESTED** [review]
(See comment about "batch" naming.)

**Pull request closed without merging**